File: src/programa/programa.dto.ts
export class ProgramaDto {
  id: string;
  nome: string;
  sigla: string;
  descricao: string;
  tipo: string;
  horarioInicio: Date;
  horarioFim: Date;
  dataInicio: Date;
  dataFim: Date;
  status: string;
  createdAt: Date;
  updatedAt: Date;
}

export interface FindAllParameters {
  nome: string;
  sigla: string;
  descricao: string;
  tipo: string;
  horarioInicio: Date;
  horarioFim: Date;
}

File: Dtos/ProgramaDto.cs
using System.ComponentModel.DataAnnotations;

public class ProgramaDto
{
    public Guid? Id { get; set; }

    [Required]
    public string Nome { get; set; }

    [Required]
    public string Sigla { get; set; }

    public string Descricao { get; set; }
    public string Tipo { get; set; }
    public DateTime HorarioInicio { get; set; }
    public DateTime HorarioFim { get; set; }
    public DateTime DataInicio { get; set; }
    public DateTime DataFim { get; set; }
    public string Status { get; set; }
}
